client/my-sites/stats/geochart/index.jsx

```diff
@@ -85,6 +99,137 @@ class StatsGeochart extends Component {
 		gaRecordEvent( 'Stats', 'Clicked Country on Map' );
 	};
 
+	// TODO: Unify this with the same function in StatsModule.
+	updateHistory( history, data ) {
```

Comment:
L103-L231 are copied from `stats-module` and we'll revisit to abstract the logic later.

cc/ @a8ck3n

Reply:
Per some Slack chatter with Jasper, we are thinking the Decorator pattern might work to unify the history/data handling. That probably means adding an HOC to wrap the `StatsModule` and the `StatsGeochart` when the `props.isRealTime` value is `true`.